H2MLauncher.UI/ViewModels/ServerFilterViewModel.cs

```diff
@@ -119,7 +173,16 @@ public bool ApplyFilter(ServerViewModel server)
                 return false;
             }
 
-            return true;
+            if (!GameModes.Any(gameMode => 
+                gameMode.IsSelected && 
+                gameMode.Model.Name.Equals(server.GameType, StringComparison.OrdinalIgnoreCase))) 
+            {
+                return false;
+            }
+
+            return MapPacks.Any(mapPackItem => 
```

Comment:
I am sure we got this wrong yesterday. This way custom maps will never show. It needs to exclude when not checked, so return false when `MapPacks.Any(item => !item.IsSelected && item.Model.Maps.Any(...))`